pkg/migration/runner.go

```diff
@@ -646,6 +670,34 @@ func (r *Runner) createCheckpointTable(ctx context.Context) error {
 	return nil
 }
 
+// registerSentinelTable sets the sentinelTable object. This is done separately
+// from creating the sentinelTable to support the case where an operator creates
+// the sentinel table manually after starting a migration.
+func (r *Runner) registerSentinelTable() {
+	sentinelName := fmt.Sprintf("_%s_sentinel", r.table.TableName)
+	r.sentinelTable = table.NewTableInfo(r.db, r.table.SchemaName, sentinelName)
+}
+
+func (r *Runner) createSentinelTable(ctx context.Context) error {
+	if r.sentinelTable == nil {
+		r.registerSentinelTable()
+	}
+	var stmt string
+	stmt = "DROP TABLE IF EXISTS %n.%n"
+	if err := dbconn.Exec(ctx, r.db, stmt, r.sentinelTable.SchemaName, r.sentinelTable.TableName); err != nil {
+		return err
+	}
+	stmt = "CREATE TABLE %n.%n (id int NOT NULL AUTO_INCREMENT PRIMARY KEY, created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP, alter_statement TEXT)"
+	if err := dbconn.Exec(ctx, r.db, stmt, r.sentinelTable.SchemaName, r.sentinelTable.TableName); err != nil {
+		return err
+	}
+	stmt = "INSERT INTO %n.%n (alter_statement) VALUES (%?)"
+	if err := dbconn.Exec(ctx, r.db, stmt, r.sentinelTable.SchemaName, r.sentinelTable.TableName, r.migration.Alter); err != nil {
+		return err
+	}
+	return nil
+}
+
```

Comment:
* Remove register
* For the dbconn.Exec, I'm trying to use the statement directly as the string argument and not via a variable. This reduces some sql injection risk because we are no longer at risk of passing around raw statements (it can be linted on in future that dbconn.Exec does not use a variable as the 3rd arg).
* For the sentinel table, we don't strictly need the alter statement. We just need the existence of the table, which you can simplify the definition of. This is because you never check the statement matches, and we more easily support the use case of manually created sentinel tables post migration start.


```suggestion
func (r *Runner) createSentinelTable(ctx context.Context) error {
	sentinelName := fmt.Sprintf("_%s_sentinel", r.table.TableName)
	if err := dbconn.Exec(ctx, r.db, "DROP TABLE IF EXISTS %n.%n",
		r.table.SchemaName,
		sentinelName); err != nil {
		return err
	}
	return dbconn.Exec(ctx, r.db, "CREATE TABLE %n.%n (id int NOT NULL AUTO_INCREMENT PRIMARY KEY, created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP)",
		r.table.SchemaName,
		sentinelName,
	)
}
```